tutorials/reddit_slack_bot/reddit_slack_bot.py

```diff
@@ -0,0 +1,169 @@
+
+# ## Creating Secrets to Access Reddit and Slack
+#
+# For our workflow to access Reddit and Slack, we need to set up some credentials
+# with Slack and Reddit, and share them with our workflow in a safe way. For Reddit,
+# we create a [developer app](https://www.reddit.com/prefs/apps) which will give
+# us a client ID and secret. We will refer to these as `reddit_client_id` and `reddit_secret_key`.
+# Once we have collected our `reddit_client_id` and `reddit_secret_key`,
+# we securely store them using the `unionai` CLI tool. To add `reddit_client_id` we run:
+# ```bash
+# unionai create secret reddit_client_id
+# ```
+# and paste the client ID when prompted.
+#
+# After, we can do the same for `reddit_secret_key`.
+
+# For Slack, we will need a Slack API bot token which can be configured and created for
+# a specific channel by following
+# [Slack's documentation](https://api.slack.com/tutorials/tracks/getting-a-token). We
+# will call this token `slack_token`.  Similar to how we added `reddit_client_id` and `reddit_secret_key`,
+# we securely store the `slack_token` in the CLI using:,
+# ```bash
+# unionai create secret slack_token
+# ```
+
+# Once our secrets are set up, we continue by importing some of the workflow dependencies
+# and setting some constants like the frequency in which we want to run the workflow and
+# the Slack channel we want to post to:
+
+import os
+from datetime import datetime, timedelta
+from typing import List, Dict
+
+from flytekit import task, workflow, ImageSpec, Secret, current_context, LaunchPlan, CronSchedule
+import requests
+from requests.auth import HTTPBasicAuth
+
+DAYS_BETWEEN_RUNS = 1
+SLACK_CHANNEL_NAME = '#reddit-posts'
+
+# ## Defining a Container Image
+#
+# The previously imported packages are either in the Python Standard Library or included by
+# default in the base flyte image used by Union. However, if we want to include additional
+# packages we define an `ImageSpec` which will create an image under the hood, so we don't
+# have to worry about writing a `Dockerfile`.
+
+image = ImageSpec(
+    registry=os.environ.get("IMAGE_SPEC_REGISTRY"),
+    packages=["slack_sdk==3.28.0"]
+)
+
+# ## Collecting Reddit Posts
+#
+# For our task, to use `reddit_client_id` and `reddit_secret_key`, we
+# specify them in the `@task` decorator using the `secret_requests` argument. Then all we need
+# to do is format a request and parst the result. We will only return posts since the last
+# time we ran the workflow; specified by `DAYS_BETWEEN_RUNS`.
+
+# We also make use of caching by setting `cache=True` in the `@task` decorator. By adding
+# `kickoff_time` and `lookback_days` as arguments to the task we keep the function code
+# independent of the current time. Now if `get_posts` ever gets called multiple times, we will
+# not make unnecessary requets to the reddit API.
+
+@task(
+    secret_requests=[
+        Secret(key="reddit_client_id"),
+        Secret(key="reddit_secret_key"),
+    ],
+    cache=True,
```

Comment:
I believe you need to add 
```suggestion
    cache=True,
    cache_version="1.0",
```